src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -841,18 +867,43 @@ void StorageReplicatedMergeTree::dropReplica(zkutil::ZooKeeperPtr zookeeper, con
     ///       "The local set of parts of table X doesn't look like the set of parts in ZooKeeper"
     ///
     {
-        Strings children = zookeeper->getChildren(remote_replica_path);
-
-        if (std::find(children.begin(), children.end(), "metadata") != children.end())
-            zookeeper->remove(fs::path(remote_replica_path) / "metadata");
+        /// Remove metadata first
+        [[maybe_unused]] auto code = zookeeper->tryRemove(fs::path(remote_replica_path) / "metadata");
+        assert(code == Coordination::Error::ZOK || code == Coordination::Error::ZNONODE);
 
-        for (const auto & child : children)
+        /// Then try to remove paths that are known to be flat (all children are leafs)
+        Strings flat_nodes = {"flags", "queue"};
```

Comment:
```suggestion
        static constexpr std::array flat_nodes = {"flags", "queue"};
```

Reply:
But it will not work with the recent changes:
https://github.com/ClickHouse/ClickHouse/blob/37a06eec1a062e5ac8ed9a179051491ea98e285c/src/Storages/StorageReplicatedMergeTree.cpp#L875-L877